core/go/pkg/persistence/gorm.go

```diff
@@ -217,3 +227,142 @@ func (gp *provider) Transaction(parentCtx context.Context, fn func(ctx context.C
 func (gp *provider) NOTX() DBTX {
 	return newNOTX(gp.gdb)
 }
+
+var (
+	whereClause     = clause.Where{}.Name()
+	postgresDialect = postgres.Dialector{}.Name()
+)
+
+// ANY is a custom implementation of the clause.IN or clause.Expr which binds array of Values directly to a single variable
+// In the clause.Expr, it replaces the IN clause with ANY, which is a PostgreSQL-specific feature.
+// it has been implemented to address the limitation of "protocol limited to 65535 parameters".
+type ANY struct {
+	IN   *clause.IN
+	Expr *clause.Expr
+}
+
+// UseAny configures the DB to use the ANY type for IN clauses to resolve parameter limitations.
+func UseAny(db *gorm.DB) {
```

Comment:
It feels to me it would be easier for Paladin code devs to wrap the value in a `db.Any(value)` using a generic function, than to wrap the `db` bit.

Or alternatively, if this code is solving the application of the pattern generally - why would this not just be an optimization that is turned on by default? (like we do for for the statement cache)

Reply:
@peterbroadhurst Needs re-review here